### PR TITLE
Feature/panel/swipe

### PR DESCRIPTION
**Pull request by gRaytan**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/131" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentStatePagerAdapter;
+
+import com.teamagam.gimelgimel.app.message.view.MessagesContainerFragment;
+import com.teamagam.gimelgimel.app.sensor.view.SensorsContainerFragment;
+
+import java.util.ArrayList;
+
+
+/**
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java]: Redundant comment

File: app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * Created by Admin on 06/12/2016.
+ */
+
+public class BottomPanelPagerAdapter extends FragmentStatePagerAdapter {
+    
+    String[] mTitles;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java]: Access modifiers missing

File: app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public int getCount() {
+        return mNumOfTitles;
+    }
+
+    @Override
+    public String getPageTitle(int position) {
+        // Generate title based on item position
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java]: Redundant comment

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java [hunk 4]
```diff
@@ -68,7 +70,9 @@
     private MainActivityNotifications mMainMessagesNotifications;
     private AlertsViewModel mAlertsViewModel;
     private MainActivityAlerts mMainActivityAlerts;
+    private MainActivityPanel mBottomPanel;
 
+    //{getString(R.string.PanelTitleSensors), getString(R.string.PanelTitleMessages)};
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java]: ?

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java [hunk 5]
```diff
@@ -210,13 +221,14 @@ private void initDrawerListener() {
     }
 
     private void initFragments() {
-        FragmentManager fragmentManager = getFragmentManager();
+
+        android.support.v4.app.FragmentManager fragmentManager = getSupportFragmentManager();
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java]: android.support.v4.app prefix is redundant

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+import java.util.ArrayList;
+import javax.inject.Inject;
+
+import butterknife.BindArray;
+import butterknife.BindString;
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: Redundant comment

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java [hunk 7]
```diff
@@ -190,10 +194,17 @@ protected int getActivityLayout() {
...
         initFragments();
+        initAlertsModule();
         initSlidingUpPanel();
+        initButtomPanel();
         initDrawerListener();
         initMainNotifications();
-        initAlertsModule();
+    }
+
+    private void initButtomPanel() {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java]: Bottom *

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    ViewPager mButtomViewPager;
+
+    @BindView(R.id.tabs)
+    PagerSlidingTabStrip tabsStrip;
+
+    @BindArray(R.array.bottom_panel_titles)
+    String[] mStringTitles;
+
+    private FragmentManager mFragmentManager;
+    private Activity mActivity;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: Field not used

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+public class MainActivityPanel implements PanelViewModel.PanelDisplayer {
+
+    @Inject
+    PanelViewModel mViewModel;
+
+    @BindView(R.id.buttom_swiping_panel)
+    ViewPager mButtomViewPager;
+
+    @BindView(R.id.tabs)
+    PagerSlidingTabStrip tabsStrip;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: Injected fields are not consistently named

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+ * Created by Admin on 13/12/2016.
+ */
+
+public class MainActivityPanel implements PanelViewModel.PanelDisplayer {
+
+    @Inject
+    PanelViewModel mViewModel;
+
+    @BindView(R.id.buttom_swiping_panel)
+    ViewPager mButtomViewPager;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: Bottom*

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+    MainActivityPanel(FragmentManager fm, Activity activity) {
+        mActivity = activity;
+        ButterKnife.bind(this, activity);
+        ((MainActivity) activity).getMainActivityComponent().inject(this);
+        mFragmentManager = fm;
+
+    }
+
+    public void Init() {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: init*

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+        mViewModel.setArguments(mStringTitles, mStringTitles.length);
+
+        BottomPanelPagerAdapter mPageAdapter =
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: local variable named as a field

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+                new BottomPanelPagerAdapter(
+                        mFragmentManager,mStringTitles,mStringTitles.length);
+        mButtomViewPager.setAdapter(mPageAdapter);
+
+         //Attach the view pager to the tab strip
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: I'd extract that line into a new method with an informative name if you'd like to keep comment information.
If this comment is not important enough for that, then it's not important enough to stay there.

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java [hunk 14]
```diff
@@ -0,0 +1,35 @@
+package com.teamagam.gimelgimel.app.mainActivity.viewmodel;
+
+import com.teamagam.gimelgimel.app.injectors.scopes.PerActivity;
+
+import java.util.ArrayList;
+
+import javax.inject.Inject;
+
+/**
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: Redundant comment

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+public class PanelViewModel {
+
+    private String[] mTitles;
+    private int mNumOfTitles;
+
+
+    @Inject
+    public PanelViewModel() {}
+
+    public void setArguments (String[] titles, int numOfTitles) {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: If the class cannot exist without these parameters, then it should ask for them at the constructor.

Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: There's no point in asking for an array and it's length.
One can simply ask the array for it's length when required.

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+        mTitles = titles;
+        mNumOfTitles = numOfTitles;
+    }
+
+    public String[] getTitles() {
+        return mTitles;
+    }
+
+    public interface PanelDisplayer {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: Seems like an empty interface.

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Inject
+    public PanelViewModel() {}
+
+    public void setArguments (String[] titles, int numOfTitles) {
+        mTitles = titles;
+        mNumOfTitles = numOfTitles;
+    }
+
+    public String[] getTitles() {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: Is anyone using that?

File: app/src/main/res/layout/activity_main_content.xml [hunk 18]
```diff
@@ -56,11 +56,30 @@
...
+            android:id="@+id/tabs"
+            app:pstsShouldExpand="true"
+            app:pstsTextAllCaps="true"
+            app:pstsIndicatorColor="@color/blue"
+            app:pstsUnderlineColor="@color/black"
+            app:pstsDividerColor="@color/black"
+            android:layout_width="match_parent"
+            android:layout_height="68dp" />
+
+        <android.support.v4.view.ViewPager xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (BarAgam) [inline app/src/main/res/layout/activity_main_content.xml]: There's no need to declare android namespace in the xml again,

File: app/src/main/res/layout/activity_main_content.xml [hunk 19]
```diff
@@ -56,11 +56,30 @@
...
+    <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:tools="http://schemas.android.com/tools"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        tools:layout="@layout/fragment_messages_container" />
+        android:orientation="vertical"
+      >
+
+        <com.astuetz.PagerSlidingTabStrip
+            android:id="@+id/tabs"
```
Comment (BarAgam) [inline app/src/main/res/layout/activity_main_content.xml]: The identifier should be more informative in such a way that will tell the developer which tabs exactly this id is referring to (main_activity_bottom_panel_tabs perhaps?)

File: app/src/main/res/layout/activity_main_content.xml [hunk 20]
```diff
@@ -56,11 +56,30 @@
...
+            app:pstsShouldExpand="true"
+            app:pstsTextAllCaps="true"
+            app:pstsIndicatorColor="@color/blue"
+            app:pstsUnderlineColor="@color/black"
+            app:pstsDividerColor="@color/black"
+            android:layout_width="match_parent"
+            android:layout_height="68dp" />
+
+        <android.support.v4.view.ViewPager xmlns:android="http://schemas.android.com/apk/res/android"
+            android:id="@+id/buttom_swiping_panel"
```
Comment (BarAgam) [inline app/src/main/res/layout/activity_main_content.xml]: bottom*

**Review by BarAgam — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java (hunk 9 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: what do you mean?

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java (hunk 10 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java (hunk 6 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java (hunk 5 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java (hunk 7 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java (hunk 4 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivity.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java (hunk 3 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java (hunk 2 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java (hunk 1 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/common/base/adapters/BottomPanelPagerAdapter.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java (hunk 8 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java (hunk 11 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java (hunk 12 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java (hunk 14 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java (hunk 15 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: i had a problem with dagger, and thats what yoni suggested. any other ideas?

Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: righe, done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java (hunk 17 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: At first, i thought we will use. i left it for future expansion

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java (hunk 16 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: It currently an empy interace. left it for future expansion

File: app/src/main/res/layout/activity_main_content.xml (hunk 19 above)
Comment (gRaytan) [reply app/src/main/res/layout/activity_main_content.xml]: ok, done

File: app/src/main/res/layout/activity_main_content.xml (hunk 18 above)
Comment (gRaytan) [reply app/src/main/res/layout/activity_main_content.xml]: done

File: app/src/main/res/layout/activity_main_content.xml (hunk 20 above)
Comment (gRaytan) [reply app/src/main/res/layout/activity_main_content.xml]: done

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java (hunk 13 above)
Comment (gRaytan) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: not sure i am agree, anyone feletd tht comment

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java (hunk 9 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: That it's not prefixed with an 'm' like the rest of the injected/binded fields

Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java]: That it's not prefixed with an 'm' like the rest of the injected/binded fields

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java (hunk 15 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/PanelViewModel.java]: Yeah. The solution to this situation is creating a factory that will inject this class' dependencies and ask for the required parameters. 
AutoFactory helps with that and generates the boilerplate code for you.
I can help you with that.

**Pull request merged**